warehouse/oidc/models/_core.py

```diff
@@ -211,7 +212,9 @@ def publisher_name(self) -> str:  # pragma: no cover
         # Only concrete subclasses are constructed.
         raise NotImplementedError
 
-    def publisher_url(self, claims=None) -> str | None:  # pragma: no cover
+    def publisher_url(
+        self, claims: SignedClaims | None = None
+    ) -> str:  # pragma: no cover
```

Comment:
I've been meaning to make this change for sometime so adding it to this PR so it's upstream sooner.  This should never be `None` right?  Or shouldn't, so types and tests should reflect that expectation?  If that's not true, I'm happy to revert.

Reply:
The original intention here is that not all publishers would necessarily have a URL to link to, so this could be `None`, although that's not currently the case.

Reply:
Ahh ok.  Then my change doesn't make sense.  Will revert.